### PR TITLE
task3-4_review

### PR DESCRIPTION
**Pull request by wooz1ewu**

### CODE REVIEW

File: task3/script/script.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+};
+
+const STATUS_COLUMN_INDEX = 2;
+
+function checkStatuses() {
+  setTimeout(() => {
+    const interval = setInterval(() => {
+      console.log("Check started..");
+
+      let table = document.querySelector("#status-table");
```
Comment (k-krasilnikova) [inline task3/script/script.js:56]: Why `let` variable used here? Where you planing to change its value?

File: task3/script/script.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+      for (let i = 0; i < table.rows.length; i++) {
+        let cell = table.rows[i].cells[STATUS_COLUMN_INDEX];
+
+        if (cell.innerText === "Pending") {
```
Comment (k-krasilnikova) [inline task3/script/script.js:61]: Let's move Done and Pending string to a separate constant and reuse it here:
```
const STATUSES = {
  pending: "Pending",
  done: "Done",
  failed: "Failed"
}
```

File: task3/script/script.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+          cell.innerText = "Done";
+          return;
```
Comment (k-krasilnikova) [inline task3/script/script.js:63]: Do we really need empty return here?

File: task3/script/script.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        }
+      }
+
+      console.log("Check finished");
+      clearInterval(interval);
+    }, 5000);
```
Comment (k-krasilnikova) [inline task3/script/script.js:69]: What is the magic numbers here? Please move them to constants at the top of file

File: task4/script/script.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+for (let d of makeDroids()) {
+  d();
+}
+
+// Task C
+
+const TIME_MULTIPLIER = 4000;
+const ACCEPTABLE_DELAY = 2000;
```
Comment (k-krasilnikova) [inline task4/script/script.js:42]: Good job with naming here!

File: task4/script/script.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+const promise = new Promise((resolve, reject) => {
+  const delay = Math.floor(Math.random() * TIME_MULTIPLIER);
+  setTimeout(() => {
+    if (delay > ACCEPTABLE_DELAY) {
+      reject(
+        "Delay " +
```
Comment (k-krasilnikova) [inline task4/script/script.js:49]: Maybe it will be easier to use `js template` here? Instead of string and +

**Review by k-krasilnikova — CHANGES_REQUESTED** [review]
Some changes required!

File: task3/script/script.js (hunk 3 above)
Comment (wooz1ewu) [reply task3/script/script.js:63]: I think we really need this empty return to stop executing of method. If we don't do that we'll execute clearInterval method, that will stop this 'cycle' of checking table fields.

**Review by k-krasilnikova — APPROVED** [review]
Approved 👍

**Pull request merged**